m1well.Utils/src/cleanup.js

```diff
@@ -0,0 +1,177 @@
+// @flow
+
+import { addDays, differenceInCalendarDays, startOfTomorrow } from 'date-fns'
+import { getOrMakeConfigurationSection } from '../../nmn.Templates/src/configuration'
+import type { Config } from './utilsHelper'
+import {
+  castNumberFromMixed,
+  castStringFromMixed,
+  futureDateFromInputOrTomorrow,
+  getDailyNote,
+  logError,
+  logMessage,
+  sortByPrio,
+  sortByType
+} from './utilsHelper'
+
+const CONFIG_KEYS = {
+  archiveNotesTag: 'archiveNotesTag',
+  archiveNotesLifeInDays: 'archiveNotesLifeInDays',
+}
+
+// if there is no config in the '_configuration' file, then provide an example config
+// currently only needed for autoArchiveNotes command
+const EXAMPLE_CONFIG = `  
+  /* >> utils plugin start <<
+   * for more information please have a look at the plugins' readme
+   */
+  utils: {
+    // just an example archiveNotesTag - please adapt to your needs
+    ${CONFIG_KEYS.archiveNotesTag}: '#scratch',
+    // here you can enter the amount of days (> 0!) of the archive lifetime notes
+    ${CONFIG_KEYS.archiveNotesLifeInDays}: 7,
+  },
+  /* >> utils plugin end << */
+`
+
+/**
+ * sort lines
+ *
+ * @private
+ */
+const sorter = async (): Promise<boolean> => {
+
+  const selectedParagraphs = Editor.selectedParagraphs
+
+  if (selectedParagraphs.length > 1) {
+    let firstIndex = 0
+
+    // check at least 2 paragraphs against all paragraphs and then get the first line index
+    // otherwise it gets inserted at lineIndex = 0
+    for (let i = 0; i < Editor.paragraphs.length; i++) {
+      if (Editor.paragraphs[i].rawContent === selectedParagraphs[0].rawContent
+        && Editor.paragraphs[i + 1].rawContent === selectedParagraphs[1].rawContent) {
+        firstIndex = Editor.paragraphs[i].lineIndex
+      }
+    }
+
+    const selectedNonTasksAndList = selectedParagraphs.filter(para => para.type !== 'open' && para.type !== 'scheduled'
```

Comment:
This is fine, but for readability, I would prefer to see something like:
const TASK_TYPES = ["open","scheduled"...
and then in both .filter statements:
TASK_TYPES.indexOf(para.type) > -1

Reply:
yes javascript is not my "mother language" :P
so i am learning the specific functions also day by day